src/awsGuard.ts

```diff
@@ -0,0 +1,184 @@
+ * Create an instance of AwsGuard without any arguments to enable all policies with their default configuration:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard();
+ * ```
+ *
+ * The above is equivalent to:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({ all: "mandatory" });
+ * ```
+ *
+ * To make all policies advisory rather than mandatory:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({ all: "advisory" });
+ * ```
+ *
+ * To make all policies advisory, but changing a couple to be mandatory:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({
+ *     all: "advisory",
+ *     ec2InstanceNoPublicIP: "mandatory",
+ *     elbAccessLoggingEnabled: "mandatory",
+ * });
+ * ```
+ *
+ * To disable a particular policy:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({
+ *     ec2InstanceNoPublicIP: "disabled",
+ * });
+ * ```
+ *
+ * To disable all policies except ones explicitly enabled:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({
+ *     all: "disabled",
+ *     ec2InstanceNoPublicIP: "mandatory",
+ *     elbAccessLoggingEnabled: "mandatory",
+ * });
+ * ```
+ *
+ * To specify configuration for policies that have it:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({
+ *     ec2VolumeInUseCheck: { enforcementLevel: "mandatory", checkDeletion: false },
+ *     encryptedVolumes: { enforcementLevel: "mandatory", kmsId: "id" },
+ * });
+ * ```
+ */
+export class AwsGuard extends PolicyPack {
+    constructor(args?: AwsGuardArgs);
+    constructor(name: string, args?: AwsGuardArgs);
+    constructor(nameOrArgs?: string | AwsGuardArgs, args?: AwsGuardArgs) {
+        const [n, a] = getNameAndArgs(nameOrArgs, args);
+        super(n, { policies: getPolicies(factoryMap, a) });
+    }
+}
+
+/**
+ * Argument bag for configuring AwsGuard policies.
+ */
+// Mixins will be applied to this interface in each module that adds a property to
+// configure each policy.
+export interface AwsGuardArgs {
+    all?: EnforcementLevel;
+}
+
+/** @internal */
+export type PolicyFactory<TArgs> = (args?: TArgs) => ResourceValidationPolicy | StackValidationPolicy;
+
+/** @internal */
+export function registerPolicy<K extends keyof AwsGuardArgs>(
+    property: Exclude<K, "all">,
+    factory: PolicyFactory<AwsGuardArgs[K]>): void {
+
+    if (typeof factory !== "function") {
+        throw new Error("'factory' must be a function.");
+    }
+    if (property === "all") {
+        throw new Error("'all' is reserved.");
+    }
+    if (property in factoryMap) {
+        throw new Error(`${property} already exists.`);
+    }
+    factoryMap[property] = factory;
+}
+
+/**
+ * Testable helper to get the name and args from the parameters,
+ * for use in the policy pack's constructor.
+ * @internal
+ */
+export function getNameAndArgs(
+    nameOrArgs?: string | AwsGuardArgs,
+    args?: AwsGuardArgs): [string, AwsGuardArgs | undefined] {
+
+    let name = defaultPolicyPackName;
+    if (typeof nameOrArgs === "string") {
```

Comment:
Would `switch (typeof nameOrArgs) {` clean up the code here? At least giving us the ability to throw for an invalid type?

Reply:
I was going to apply your suggestion, but it ended up complicating things more. Certainly could do some more arg validation here, but I will look at it subsequently.